packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -186,7 +185,14 @@ async function getSelectionsFromContributors<T>(
         name: 'selection',
         message: selectionOptions.selectionPrompt,
         choices: selections,
-        default: selectionOptions.listOptionsField === 'runtimes' ? 'nodejs' : undefined,
+        default:
+          selectionOptions.service === ServiceName.LambdaFunction
+            ? selectionOptions.listOptionsField === 'runtimes'
+              ? 'nodejs'
+              : undefined
+            : selectionOptions.runtimeState === undefined
+            ? undefined
+            : prevruntime,
```

Comment:
This should be its own function and use if statements rather than ternary

Reply:
done